memory_set/src/set.rs

```diff
@@ -29,18 +31,18 @@ impl<F: Copy, P, B: MappingBackend<F, P>> MemorySet<F, P, B> {
     }
 
     /// Returns the iterator over all memory areas.
-    pub fn iter(&self) -> impl Iterator<Item = &MemoryArea<F, P, B>> {
+    pub fn iter(&self) -> impl Iterator<Item = &MemoryArea<B>> {
         self.areas.values()
     }
 
     /// Returns whether the given address range overlaps with any existing area.
-    pub fn overlaps(&self, range: VirtAddrRange) -> bool {
-        if let Some((_, before)) = self.areas.range(..range.start).last() {
+    pub fn overlaps(&self, range: AddrRange<B::Addr>) -> bool {
+        if let Some((_, before)) = self.areas.range(..range.start.into()).last() {
```

Comment:
> When MemoryAddr impl Ord, can B::Addr be used as a key here?

So we can abandon this ugly `.into()`

Reply:
Yes

Reply:
Then I think `Ord` for `MemoryAddr` is an necessity. And maybe it can be updated in this PR together?

I think we've talk about this before? @equation314

Reply:
> Then I think `Ord` for `MemoryAddr` is an necessity. And maybe it can be updated in this PR together?

I've pushed a version to #4.